lib/resolve-config.js

```diff
@@ -40,5 +43,40 @@ module.exports = (
         ? CI_API_V4_URL
         : urlJoin(defaultedGitlabUrl, isNil(userGitlabApiPathPrefix) ? '/api/v4' : userGitlabApiPathPrefix),
     assets: assets ? castArray(assets) : assets,
+    proxy: getProxyConfiguration(userGitlabUrl, HTTP_PROXY, HTTPS_PROXY),
```

Comment:
userGitlabUrl might not be defined.  So I think
```suggestion
    proxy: getProxyConfiguration(defaultedGitlabUrl, HTTP_PROXY, HTTPS_PROXY),
```
should work.